executor/src/witgen/mod.rs

```diff
@@ -123,9 +123,181 @@ fn rows_are_repeating(values: &[(&str, Vec<FieldElement>)]) -> Option<usize> {
     })
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum CompleteCause {
```

Comment:
@chriseth these are all the cases where we should ignore this identity the next time for a specific reason. If all of these are compiler errors. `None` is the happy case here, we solved everything we could and don't have anything to report. Do you think we should just panic instead of bubbling these up? They signal limitations of the executor.

Reply:
Yes, I think panicking would be better. If the identity is there, it is there for a reason and we probably won't be able to generate the witnesses an way.

Reply:
Done!